### PR TITLE
MAYH-9336 antiope-sns

### PR DESCRIPTION
**Pull request by jianwan2016**
* Added `antiope-sns`
* Added warnings and cleaned.

Test:
```
λ> env <- mkEnv Oregon (\l bs -> return ()) 
λ> runResourceT $ runAWS env $ publishMsg "arn:aws:sns:us-west-2:143032791481:wan-MaxMindRawTopic" "test"
it :: ()
```

And run 10s of `publishMsg`, check `lsof` to see whether there are leaked connection: looks like no.
```
λ> sequence_ $ [runResourceT $ runAWS env $ publishMsg "arn:aws:sns:us-west-2:143032791481:wan-MaxMindRawTopic" "test" | _ <- take 1000 [1..]]
jwan@~/Start/atlas/python $ lsof -p 38158  | wc -l
     153
jwan@~/Start/atlas/python $ lsof -p 38158  | wc -l
     153
```

```
runResourceT $ runAWS env $ sequence_ $ [publishMsg "arn:aws:sns:us-west-2:143032791481:wan-MaxMindRawTopic" "test" | _ <- take 1000 [1..]]

jwan@~/Start/atlas/python $ lsof -p 38158  | wc -l
     153
```

### CODE REVIEW

File: antiope-sns/src/Antiope/SNS.hs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+) where
+
+import Control.Lens
+import Data.Text       (Text)
+import Network.AWS     (MonadAWS, send)
+import Network.AWS.SNS
+
+publishMsg :: MonadAWS m => Text -> Text -> m ()
+publishMsg topicArn message = do
+  _ <- send $ publish message & pTopicARN ?~ topicArn
```
Comment (dsturnbull) [inline antiope-sns/src/Antiope/SNS.hs]: If we don't consume the response, then I think we could hit the unresolved `ResourceT` issue.

To test, send a few thousand messages and inspect the process with `lsof` and look for thousands of left-over connections.

File: antiope-athena/src/Antiope/Athena.hs [hunk 2]
```diff
@@ -41,8 +27,8 @@ query config context qstring clientRequestToken = do
     Nothing   -> return []
   where
     go qeid = do
-      accept <- await queryExecutionFinished (getQueryExecution qeid)
-      case accept of
+      accpt <- await queryExecutionFinished (getQueryExecution qeid)
+      case accpt of
```
Comment (newhoggy) [inline antiope-athena/src/Antiope/Athena.hs]: The extra `e` isn't that bad is it? 😁

File: antiope-sns/src/Antiope/SNS.hs (hunk 1 above)
Comment (newhoggy) [reply antiope-sns/src/Antiope/SNS.hs]: Good point.

File: antiope-athena/src/Antiope/Athena.hs (hunk 2 above)
Comment (jianwan2016) [reply antiope-athena/src/Antiope/Athena.hs]: symbol shadowing complained by compiler :D

File: antiope-sns/src/Antiope/SNS.hs (hunk 1 above)
Comment (jianwan2016) [reply antiope-sns/src/Antiope/SNS.hs]: How did we fix that issue before?

Comment (dsturnbull) [reply antiope-sns/src/Antiope/SNS.hs]: We consumed the response :)

i.e. `ackMessage -> m` in SQS is now defined as `ackMessages [m]`, which reads the responses through the batch api.

File: antiope-athena/src/Antiope/Athena.hs (hunk 2 above)
Comment (dsturnbull) [reply antiope-athena/src/Antiope/Athena.hs]: Can we hide the offender instead of renaming?

Comment (newhoggy) [reply antiope-athena/src/Antiope/Athena.hs]: It's probably better to use `hiding` or qualified imports in this case.

File: antiope-dynamodb/package.yaml [hunk 3]
```diff
@@ -44,6 +44,15 @@ dependencies:
...
 - unordered-containers
 
+ghc-options:
+- -Wall
+- -Wcompat
+- -Wincomplete-record-updates
+- -Wincomplete-uni-patterns
+- -Wredundant-constraints
+- -O2
+- -msse4.2
```
Comment (newhoggy) [inline antiope-dynamodb/package.yaml]: I like the stricter options.

File: antiope-athena/src/Antiope/Athena.hs (hunk 2 above)
Comment (jianwan2016) [reply antiope-athena/src/Antiope/Athena.hs]: Fixed, thanks.

File: antiope-sns/src/Antiope/SNS.hs (hunk 1 above)
Comment (jianwan2016) [reply antiope-sns/src/Antiope/SNS.hs]: Thanks.

**Pull request merged**